client/state/jetpack-agency-dashboard/actions.ts

```diff
@@ -88,3 +94,45 @@ export function setSiteMonitorStatus( siteId: number, status: 'loading' | 'compl
 		status,
 	};
 }
+
+export function setSiteValidatingStatusValidating( siteUrl: string ) {
+	return {
+		type: JETPACK_AGENCY_DASHBOARD_CONNECT_URL_SITE_VALIDATION_VALIDATING,
+		siteUrl,
+	};
+}
+
+export function setSiteValidatingStatusJetpackConnected( siteUrl: string ) {
+	return {
+		type: JETPACK_AGENCY_DASHBOARD_CONNECT_URL_SITE_VALIDATION_JETPACK_CONNECTED,
+		siteUrl,
+	};
+}
+
+export function setSiteValidatingStatusWordPressSite( siteUrl: string ) {
+	return {
+		type: JETPACK_AGENCY_DASHBOARD_CONNECT_URL_SITE_VALIDATION_WORDPRESS_SITE,
+		siteUrl,
+	};
+}
+
+export function setSiteValidatingStatusNonWordPressSite( siteUrl: string ) {
+	return {
+		type: JETPACK_AGENCY_DASHBOARD_CONNECT_URL_SITE_VALIDATION_NON_WORDPRESS_SITE,
+		siteUrl,
+	};
+}
+
+export function setSiteValidatingStatusNotExists( siteUrl: string ) {
+	return {
+		type: JETPACK_AGENCY_DASHBOARD_CONNECT_URL_SITE_VALIDATION_NOT_EXISTS,
+		siteUrl,
+	};
+}
+
+export function setSiteValidatingStatusError( siteUrl: string ) {
+	return {
+		type: JETPACK_AGENCY_DASHBOARD_CONNECT_URL_SITE_VALIDATION_ERROR,
+		siteUrl,
+	};
+}
```

Comment:
The following may not matter depending on whether we drop the validation step:
For something isolated like submitting a list of URLs that does not affect the general state of Calypso I would advise against using redux. For something as isolated and simple, useState and useContext should be more than sufficient (and far easier to maintain).

Reply:
This was a tricky one, that I also didn't wat to resort to redux, but using `useState` on the `ConnectUrl` component, would trigger component renders from a child component ( `QueryJetpackManageAddSiteUrl` ) at the `onSuccess` callback. This would make React crash, so I resorted to redux here.